desktop/api/src/main/java/org/datacleaner/actions/PreviewTransformedDataActionListener.java

```diff
@@ -290,6 +290,14 @@ private void sanitizeIrrelevantComponents(AnalysisJobBuilder ajb, TransformerCom
                     // components
                     relevantAnalysisJobBuilder.removeComponent(componentBuilder);
                 }
+
+                if (!importantComponent) {
+                    // remove the components that are not configured.
+                    if (componentBuilder.getInput().length == 0) {
```

Comment:
We could do this more generically by asking if componentBuilder is configured. Then it would not be restricted to just input columns, but to any missing configuration aspect.